packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -959,7 +934,124 @@ where
+                        entity_type_version_id,
+                        properties,
+                        left_order, right_order,
+                        left_entity_id, right_entity_id,
+                        owned_by_id, created_by_id, updated_by_id
+                ),
+                inserted_in_historic AS (
+                    -- We immediately put this deleted entity into the historic table.
+                    -- As this should be done in a transaction, we should be safe that this move
+                    -- doesn't produce invalid state.
+                    INSERT INTO entity_histories(
+                        entity_id, version,
+                        entity_type_version_id,
+                        properties,
+                        left_order, right_order,
+                        left_entity_id, right_entity_id,
+                        owned_by_id, created_by_id, updated_by_id, 
+                        archived
+                    )
+                    SELECT
+                        entity_id, version,
+                        entity_type_version_id,
+                        properties,
+                        left_order, right_order,
+                        left_entity_id, right_entity_id,
+                        owned_by_id, created_by_id, updated_by_id,
+                        $2::boolean
+                    FROM to_move_to_historic
+                    -- For metadata purposes, we return the entire entity
+                    RETURNING *
+                )
+                SELECT
+                    entity_id, inserted_in_historic.version,
+                    base_uri, type_ids.version,
+                    properties, -- Unused
+                    left_order, right_order,
+                    left_entity_id, right_entity_id,
+                    owned_by_id, created_by_id, updated_by_id
+                FROM inserted_in_historic
+                INNER JOIN type_ids ON inserted_in_historic.entity_type_version_id = type_ids.version_id;
+                "#,
+                &[
+                    &entity_id,
+                    &archived,
+                ],
+            )
+            .await
+            .into_report()
+            .change_context(InsertionError)?;
+
+        let link_metadata = match (
+            historic_entity.get(5),
+            historic_entity.get(6),
+            historic_entity.get(7),
+            historic_entity.get(8),
+        ) {
+            (left_order, right_order, Some(left_entity_id), Some(right_entity_id)) => Some(
+                LinkEntityMetadata::new(left_entity_id, right_entity_id, left_order, right_order),
+            ),
+            _ => None,
+        };
+
+        let base_uri = BaseUri::new(historic_entity.get(2))
+            .into_report()
+            .change_context(InsertionError)?;
+        let entity_type_id = VersionedUri::new(base_uri, historic_entity.get::<_, i64>(3) as u32);
+
+        // TODO: Consider using `ProvenanceMetadata` instead?
```

Comment:
It was more if we could get away with not even using the top level metadata struct in the first place, but only use a lighter provenancemetadata struct instead of it. May not make as much sense to do with the method being shared between update and archive, though, as the `link_metadata` is extracted by the update method. I'll remove this todo

Reply:
Gotcha, I don't think the provenance metadata struct would be good enough. We want to return the identifier for example ("here's your new version")

Reply:
Removed todo in 909ae060a6ded6b905bfad35dac21981709d5dbc